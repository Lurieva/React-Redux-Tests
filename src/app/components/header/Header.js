import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button, LogoName } from '../../shared';

import './header.scss';

const Header = ({ isShowSearchBtn, children }) => (
    <header className="header">
        <div className="header__logo">
            <LogoName />
                { 
                    isShowSearchBtn &&  
                    <Button className="header__search-button" 
                            name='Search'
                            onClick={() => window.location = '/'}>
                    </Button>
                }
        </div>
        {children}
    </header> 
);

Header.propTypes = {
    children: PropTypes.element.isRequired
};

export default Header;