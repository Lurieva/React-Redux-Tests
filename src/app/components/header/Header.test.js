import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

jest.mock('../../shared/Button/Button', () => 'Button');
jest.mock('../../shared/LogoName/LogoName', () => 'LogoName');

describe('Header', () => {
    it('should be defined', () => {
        expect(Header).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<Header isShowSearchBtn={true} />);
        expect(tree).toMatchSnapshot();
    });

    it('should render button', () => {
        const tree = shallow(<Header isShowSearchBtn={true} />);
        expect(tree.children().length).toBe(1);
    });
});