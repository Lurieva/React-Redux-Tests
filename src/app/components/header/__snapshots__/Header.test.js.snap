// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isShowSearchBtn={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header__logo"
        >
          <LogoName />
          <Button
            className="header__search-button"
            name="Search"
            onClick={[Function]}
          />
        </div>,
        undefined,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <LogoName />,
            <Button
              className="header__search-button"
              name="Search"
              onClick={[Function]}
            />,
          ],
          "className": "header__logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "LogoName",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header__search-button",
              "name": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "Button",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__logo"
          >
            <LogoName />
            <Button
              className="header__search-button"
              name="Search"
              onClick={[Function]}
            />
          </div>,
          undefined,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <LogoName />,
              <Button
                className="header__search-button"
                name="Search"
                onClick={[Function]}
              />,
            ],
            "className": "header__logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "LogoName",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header__search-button",
                "name": "Search",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "Button",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
